### PR TITLE
Prevent done during reset

### PR DESCRIPTION
**Pull request by liu-allan**
Disables `dones` from returning true during a reset

#1258

### CODE REVIEW

File: smarts/core/sensors.py [hunk 1]
```diff
@@ -458,15 +458,16 @@ def _is_done_with_events(cls, sim, agent_id, vehicle, sensor_state):
         )
 
         done = (
-            (is_off_road and done_criteria.off_road)
+            (not sim._resetting) and
```
Comment (sah-huawei) [inline smarts/core/sensors.py:461]: It would  be better to use a non-private property here, i.e., to promote `_resetting` to `resetting` in the Smarts class.

**Review by sah-huawei — COMMENTED** [review]
I'm not sure why the CI tests are failing.  It seems to be something different than #1230.  We can probably ignore that here, but it might be good for more than one of us to double-check that before we merge this.

**Comment by Gamenot** [comment]
#1267 will need to be merged for the tests to succeed here.

File: smarts/core/sensors.py [hunk 2]
```diff
@@ -458,15 +458,16 @@ def _is_done_with_events(cls, sim, agent_id, vehicle, sensor_state):
         )
 
         done = (
-            (is_off_road and done_criteria.off_road)
+            (not sim.resetting) and
```
Comment (Gamenot) [inline smarts/core/sensors.py:461]: nit:
```suggestion
            not sim.resetting and 
```

**Review by Gamenot — APPROVED** [review]

**Review by sah-huawei — APPROVED** [review]

File: smarts/core/smarts.py [hunk 3]
```diff
@@ -954,7 +954,7 @@ def providers(self) -> List[Provider]:
         # TODO: It's inconsistent that pybullet is not here
         return self._providers
 
-    def get_provider_by_type(self, requested_type) -> Provider:
+    def get_provider_by_type(self, requested_type):
```
Comment (liu-allan) [inline smarts/core/smarts.py:957]: @Gamenot Can you look over these changes in `smarts.py`? This was to fix pytype errors when it would return `None` instead of `Provider`

Comment (sah-huawei) [inline smarts/core/smarts.py:957]: I would suggest this instead be:  `-> Optional[Provider]`

File: smarts/core/smarts.py [hunk 4]
```diff
@@ -994,7 +994,7 @@ def _reset_providers(self):
             except Exception as provider_error:
                 self._handle_provider(provider, provider_error)
 
-    def _handle_provider(self, provider: Provider, provider_error) -> ProviderState:
+    def _handle_provider(self, provider: Provider, provider_error):
         provider_problem = bool(provider_error or not provider.connected)
         if not provider_problem:
             return
```
Comment (sah-huawei) [inline smarts/core/smarts.py:1000]: Same with for this one (`-> Optional[ProviderState]`).  But also, I think we actually should specify `return None` on line 1000.

File: smarts/core/smarts.py [hunk 5]
```diff
@@ -1127,6 +1127,11 @@ def should_reset(self):
         """If the simulation requires a reset."""
         return self._reset_required
 
+    @property
+    def resetting(self):
```
Comment (Gamenot) [inline smarts/core/smarts.py:1131]: Nit since it is evident:
```suggestion
    def resetting(self) -> bool:
```

File: smarts/core/smarts.py [hunk 6]
```diff
@@ -1380,7 +1385,7 @@ def _try_emit_envision_state(self, provider_state: ProviderState, obs, scores):
                 )
                 traffic[v.vehicle_id] = envision_types.TrafficActorState(
                     actor_type=envision_types.TrafficActorType.SocialVehicle,
-                    vehicle_type=veh_type,
+                    vehicle_type=envision_types.VehicleType(veh_type),
```
Comment (sah-huawei) [inline smarts/core/smarts.py:1388]: We will probably need to add a function to convert the types here.
However, I would suggest that it should be done in a different PR.

File: smarts/core/smarts.py (hunk 3 above)
Comment (Gamenot) [reply smarts/core/smarts.py:957]: I have the same opinion as @sah-huawei

File: smarts/core/smarts.py (hunk 6 above)
Comment (sah-huawei) [reply smarts/core/smarts.py:1388]: (... the reason I suggest a different PR is because we've long needed to clean up our handling of vehicle types anyway and it's a bigger job than just this.)

Comment (liu-allan) [reply smarts/core/smarts.py:1388]: Is what is here okay for now though?

**Comment by liu-allan** [comment]
> #1267 will need to be merged for the tests to succeed here.

It works, 1267 should be merged before this

File: smarts/core/smarts.py (hunk 6 above)
Comment (liu-allan) [reply smarts/core/smarts.py:1388]: @sah-huawei can you take another look at this please?

File: envision/types.py [hunk 7]
```diff
@@ -38,7 +38,7 @@ class VehicleType(str, Enum):
     Coach = "coach"
     Truck = "truck"
     Trailer = "trailer"
-    Car = "car"
+    Passenger = "passenger"
```
Comment (sah-huawei) [inline envision/types.py:41]: This should be reverted too.  
Note that the list here in `envison/types` will not match the union of the two possible type lists that we currently support (the keys to the `VEHICLE_CONFIG` dict in `vehicle.py` and then the values of the `vehicle_type` fields in the `VehicleConfig` objects in that dict).

To fix this (in another PR), we will either need to add a function to map from the bigger list into the list of 5 things that envision supports, or (probably better, but non-trivial) unify across our vehicle typing schemes.

File: smarts/core/smarts.py [hunk 8]
```diff
@@ -1350,7 +1355,7 @@ def _try_emit_envision_state(self, provider_state: ProviderState, obs, scores):
                 traffic[v.vehicle_id] = envision_types.TrafficActorState(
                     name=self._agent_manager.agent_name(agent_id),
                     actor_type=actor_type,
-                    vehicle_type=envision_types.VehicleType.Car,
+                    vehicle_type=envision_types.VehicleType.Passenger,
```
Comment (sah-huawei) [inline smarts/core/smarts.py:1358]: When the above is reverted, don't forget to revert this too.

File: smarts/core/smarts.py (hunk 6 above)
Comment (sah-huawei) [reply smarts/core/smarts.py:1388]: As described above, and as we just chatted about, getting this right is going to take more than this.  Thus, I think it's best to just revert this line here to what it was originally.  I will open another issue about fixing vehicle types in general.

**Review by sah-huawei — CHANGES_REQUESTED** [review]
Everything is fine w.r.t. the original point of this PR, I just ask that you revert the changes related to vehicle types.
